apps/SageAccountWeb/src/App.tsx

```diff
@@ -1,37 +1,77 @@
+  SynapseClient,
+  SynapseContextConsumer,
+  SynapseContextType,
+  SynapseQueries,
+  useSynapseContext,
+} from 'synapse-react-client'
+import { useAppContext } from './AppContext'
 import { AccountCreatedPage } from './components/AccountCreatedPage'
 import { AccountSettings } from './components/AccountSettings'
 import { CertificationQuiz } from './components/CertificationQuiz'
+import { ChangePasswordPage } from './components/ChangePasswordPage'
 import { CurrentAffiliationPage } from './components/CurrentAffiliationPage'
+import Footer from './components/Footer'
 import { JoinTeamPage } from './components/JoinTeamPage'
+import { OAuthClientManagementPage } from './components/OAuthClientManagementPage'
+import { PersonalAccessTokensPage } from './components/PersonalAccessTokensPage'
 import { ProfileValidation } from './components/ProfileValidation/ProfileValidation'
 import { RegisterAccount1 } from './components/RegisterAccount1'
 import { RegisterAccount2 } from './components/RegisterAccount2'
 import { ResetPassword } from './components/ResetPassword'
+import { SageResourcesPage } from './components/SageResourcesPage'
+import { SignUpdatedTermsOfUsePage } from './components/SignUpdatedTermsOfUsePage'
 import { TermsOfUsePage } from './components/TermsOfUsePage'
-import React from 'react'
-import { Route, Switch } from 'react-router-dom'
-import {
-  CookiesNotification,
-  SynapseClient,
-  SynapseContextConsumer,
-  SynapseContextType,
-  processRedirectURLInOneSage,
-} from 'synapse-react-client'
+import { ResetTwoFactorAuth } from './components/TwoFactorAuth/ResetTwoFactorAuth'
+import TwoFactorAuthBackupCodesPage from './components/TwoFactorAuth/TwoFactorAuthBackupCodesPage'
+import TwoFactorAuthEnrollmentPage from './components/TwoFactorAuth/TwoFactorAuthEnrollmentPage'
 import { WebhookManagementPage } from './components/WebhooksManagementPage'
+import { RESET_2FA_ROUTE } from './Constants'
+import LoginPage from './LoginPage'
 import { getSearchParam } from './URLUtils'
 import './App.scss'
-import LoginPage from './LoginPage'
-import TwoFactorAuthEnrollmentPage from './components/TwoFactorAuth/TwoFactorAuthEnrollmentPage'
-import TwoFactorAuthBackupCodesPage from './components/TwoFactorAuth/TwoFactorAuthBackupCodesPage'
-import { PersonalAccessTokensPage } from './components/PersonalAccessTokensPage'
-import { OAuthClientManagementPage } from './components/OAuthClientManagementPage'
-import { ResetTwoFactorAuth } from './components/TwoFactorAuth/ResetTwoFactorAuth'
-import { RESET_2FA_ROUTE } from './Constants'
-import { ChangePasswordPage } from './components/ChangePasswordPage'
-import { SignUpdatedTermsOfUsePage } from './components/SignUpdatedTermsOfUsePage'
+
+function LoggedInRedirector() {
+  const { accessToken } = useSynapseContext()
+  const appContext = useAppContext()
+  const { data: tosStatus } = SynapseQueries.useTermsOfServiceStatus(
+    accessToken,
+    {
+      enabled: !!accessToken,
+    },
+  )
```

Comment:
Add a check on the ToS. If you MUST agree to the ToS, send you to the ToS page instead of the app you tried to log into

Reply:
Not sure why the AppInitializer is not properly handling, there must be a race condition when accessing the root.  Thanks for adding this!

Reply:
Slipped my mind that you added this logic to the AppInitializer. I'm going to remove my redirect and let it handle yours, but keep the logic to skip the race condition.